zarr/storage.py

```diff
@@ -54,6 +53,24 @@
     from zarr.codecs import Zlib
     default_compressor = Zlib()
 
+# Find the best function for FS replace.
+# Preferably atomic.
+if hasattr(os, 'replace'):
+    # Python >= 3.3 has replace() which can replace existing
+    # files on all platforms.
+    replace = os.replace
+else:
+    # In Python < 3.3 use rename().
+    if os.name == 'nt':
+        def replace(old, new):
+            if os.path.exists(new):
+                # On windows, rename() can't overwrite files. So
+                # the file is removed first.
+                os.remove(new)
+            os.rename(old, new)
```

Comment:
So this is the case where we would want the `pyosreplace` package. It's a Windows only backport for `os.replace` on Python 2.